### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ik-web**

### CODE REVIEW

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,53 @@
...
+      newFormat.push(year);
+    }
+
+    if (el[0] === 'M') {
+      newFormat.push(month);
+    }
+
+    if (el[0] === 'D') {
+      newFormat.push(day);
+    }
```
Comment (illiarerun) [inline src/formatDate.js:96]: follow DRY - don't repeat yourself, you basically do the same operation in this 2 if statements, so combine them

File: src/formatDate.js [hunk 2]
```diff
@@ -50,7 +50,53 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let year = '';
+  let month = '';
+  let day = '';
+
+  const oldFormat = date.split(fromFormat[3]);
```
Comment (illiarerun) [inline src/formatDate.js:57]: It'd be better to create variable for `fromFormat[3]`, e.g. `oldSeparator`
```suggestion
  const oldFormat = date.split(fromFormat[3]);
```

File: src/formatDate.js [hunk 3]
```diff
@@ -50,7 +50,53 @@
...
+    if (el[0] === 'M') {
+      newFormat.push(month);
+    }
+
+    if (el[0] === 'D') {
+      newFormat.push(day);
+    }
+  });
+
+  return newFormat.join(toFormat[3]);
```
Comment (illiarerun) [inline src/formatDate.js:99]: same here - `newSeparator`

File: src/formatDate.js [hunk 4]
```diff
@@ -50,7 +50,53 @@
...
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let year = '';
+  let month = '';
+  let day = '';
+
+  const oldFormat = date.split(fromFormat[3]);
+
+  fromFormat.forEach((el, i) => {
```
Comment (illiarerun) [inline src/formatDate.js:59]: smth like `datePart` would be more fitting
```suggestion
  fromFormat.forEach((datePart, i) => {
```

File: src/formatDate.js [hunk 5]
```diff
@@ -50,7 +50,53 @@
...
+    }
+
+    if (el[0] === 'D') {
+      day = oldFormat[i];
+    }
+  });
+
+  const newFormat = [];
+
+  toFormat.forEach((el) => {
```
Comment (illiarerun) [inline src/formatDate.js:75]: same here

File: src/formatDate.js [hunk 6]
```diff
@@ -50,7 +50,53 @@
...
+    if (el[0] === 'Y') {
+      if (year.length > el.length) {
+        year = year.slice(2);
+      } else if (year.length < el.length) {
+        if (+year < 30) {
+          year = '20' + year;
+        } else {
+          year = '19' + year;
+        }
```
Comment (illiarerun) [inline src/formatDate.js:84]: you can create a variable for century value, so you wouldn't need to repeat yourself with `year = n + year`, you'll do it only once at the end

Comment (illiarerun) [reply src/formatDate.js:84]: and also it'd be great to use ternary for that)

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ , now let's work on some fixes:

File: src/formatDate.js (hunk 2 above)
Comment (ik-web) [reply src/formatDate.js]: Fixed

File: src/formatDate.js (hunk 3 above)
Comment (ik-web) [reply src/formatDate.js]: Fixed

File: src/formatDate.js (hunk 6 above)
Comment (ik-web) [reply src/formatDate.js]: Fixed
Thanks for your help once more!

**Pull request closed without merging**